core/lib/eth_signer/src/json_rpc_signer.rs

```diff
@@ -380,43 +384,40 @@ mod tests {
         })
     }
     #[derive(Clone)]
-    struct State {
+    struct ServerState {
         private_keys: Vec<H256>,
     }
 
-    fn run_server(state: State) -> (String, AbortHandle) {
+    async fn run_server(state: ServerState) -> (String, AbortHandle) {
         let mut url = None;
         let mut server = None;
+        let app = Router::new()
+            .route("/", post(index))
+            .with_state(Arc::new(state));
+
         for i in 9000..9999 {
             let new_url = format!("127.0.0.1:{}", i);
-            // Try to bind to some port, hope that 999 variants will be enough
-            let tmp_state = state.clone();
-            if let Ok(ser) = HttpServer::new(move || {
-                App::new()
-                    .app_data(Data::new(tmp_state.clone()))
-                    .service(index)
-            })
-            .bind(new_url.clone())
-            {
-                server = Some(ser);
+            if let Ok(listener) = tokio::net::TcpListener::bind(&new_url).await {
+                server = Some(axum::serve(listener, app));
                 url = Some(new_url);
                 break;
             }
         }
 
         let server = server.expect("Could not bind to port from 9000 to 9999");
         let (abort_handle, abort_registration) = AbortHandle::new_pair();
-        let future = Abortable::new(server.run(), abort_registration);
+        let future = Abortable::new(server.into_future(), abort_registration);
```

Comment:
Probably not idiomatic, but it was quicker than trying to implement an idiomatic graceful shutdown, and probably it's not worth it given that it's a unit test for the code we don't even use (yet).